db/crud.go

```
@@ -358,10 +351,16 @@ func (db *Database) GetDelta(docID, fromRevID, toRevID string) (delta *RevisionD
 	if fromRevision.Body != nil {
 
 		db.DbStats.StatsDeltaSync().Add(base.StatKeyDeltaCacheMisses, 1)
-		toBody, err := db.revisionCache.Get(docID, toRevID, BodyDeepCopy)
+		toRevision, err := db.revisionCache.Get(docID, toRevID, BodyDeepCopy)
 		if err != nil {
-			return nil, err
+			return nil, nil, err
 		}
+
```

I think we should be checking for toRevision.body=nil here, and returning a 404 missing if body=nil and err=nil (same as GetRev).

Looking at it closer, is there a bug in GetRev when revisionCache.Get returns error?  Seems like it's ignoring the error if the body isn't nil.

---

Agreed to leave this as-is, as it's enough of a corner case that the overhead of the additional Get isn't worth the additional code complexity.